src/state-mutation.js

```diff
@@ -178,3 +178,10 @@ export const setIsInvalidatingForExistingResource = (state, { type, id }, value
     ? imm(state).del(updatePath)
     : imm(state).set(updatePath, value);
 };
+
+export const insertResourcesTypeIntoState = (state, type) => {
+  const path = [type, 'data'];
+  return state[type]
+    ? state
+    : imm(state).set(path, []).value();
```

Comment:
You could utilize `hasOwnProperties` from `utils` to check the whole path, instead of just checked that `state` has `type`.